src/main/java/io/temporal/samples/hello/HelloSideEffect.java

```diff
@@ -0,0 +1,266 @@
+       * On replay the same value is returned so determinism is guaranteed.
+       */
+      sideEffectsRandomID = Workflow.sideEffect(String.class, UUID.randomUUID()::toString);
+      savedSideEffectsRandomID = Workflow.sideEffect(String.class, () -> sideEffectsRandomID);
+
+      /**
+       * Let's now show unsafe (not-deterministic way) If we use the new Random().nextInt() it will
+       * be re-evaluated on replay However savedUnsafeRandomInt use sideEffect and will have the
+       * save return value on replay.
+       */
+      unsafeRandomInt = new Random().nextInt();
+      savedUnsafeRandomInt = Workflow.sideEffect(int.class, () -> unsafeRandomInt);
+
+      /**
+       * Execute activity (sync) - with sticky exec disabled (see WorkerFactory creation in main)
+       */
+      String result = activities.doSomething("Hello");
+
+      if (randomInt != savedRandomInt) {
+        throw new IllegalStateException("Random ints not equal");
+      }
+
+      if (sideEffectsRandomInt != savedSideEffectsRandomInt) {
+        throw new IllegalStateException("Sideffect random ints not equal");
+      }
+
+      if (!randomID.equals(savedRandomID)) {
+        throw new IllegalStateException("Random IDs not equal");
+      }
+
+      if (!sideEffectsRandomID.equals(savedSideEffectsRandomID)) {
+        throw new IllegalStateException("Sideffect random IDs not equal");
+      }
+
+      // This check will fail as unsafeRandomInt it was created not safely
+      // and breaks workflow determinism
+      if (unsafeRandomInt != savedUnsafeRandomInt) {
+        return "Unsafely created random numbers are not equal";
+      }
+
+      return result;
+    }
+  }
+
+  /** Simple activity implementation. */
+  static class SideEffectActivitiesImpl implements SideEffectActivities {
+    @Override
+    public String doSomething(String input) {
+      return "Received input: " + input;
+    }
+  }
+
+  /**
+   * With our Workflow and Activities defined, we can now start execution. The main method starts
+   * the worker and then the workflow.
+   */
+  public static void main(String[] args) {
+
+    // Define the workflow service.
+    WorkflowServiceStubs service = WorkflowServiceStubs.newInstance();
+
+    /*
+     * Define the workflow client. It is a Temporal service client used to start, signal, and query
+     * workflows
+     */
+    WorkflowClient client = WorkflowClient.newInstance(service);
+
+    /*
+     * Define the workflow factory. It is used to create workflow workers for a specific task queue.
+     * For this example we set the setWorkflowHostLocalTaskQueueScheduleToStartTimeout to zero
+     * (to disable sticky execution)
+     */
+    WorkerFactory factory =
+        WorkerFactory.newInstance(
+            client,
+            WorkerFactoryOptions.newBuilder()
+                .setWorkflowHostLocalTaskQueueScheduleToStartTimeout(Duration.ZERO)
```

Comment:
Never set this in the samples. It is purely for unit testing SDK.

In samples, you can complete a workflow and then query it which is going to force the replay.

Reply:
makes sense. updated